testing/camunda-process-test-java/src/main/java/io/camunda/process/test/impl/mock/BpmnExampleDataReader.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.process.test.impl.mock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.client.CamundaClient;
+import io.camunda.process.test.api.CamundaAssertAwaitBehavior;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.model.bpmn.impl.ZeebeConstants;
+import io.camunda.zeebe.model.bpmn.instance.BaseElement;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeProperties;
+import io.camunda.zeebe.model.bpmn.instance.zeebe.ZeebeProperty;
+import java.io.ByteArrayInputStream;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import org.awaitility.core.ConditionTimeoutException;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+
+public class BpmnExampleDataReader {
+
+  private final CamundaClient client;
+  private final CamundaAssertAwaitBehavior awaitBehavior;
+
+  public BpmnExampleDataReader(
+      final CamundaClient client, final CamundaAssertAwaitBehavior awaitBehavior) {
+
+    this.client = client;
+    this.awaitBehavior = awaitBehavior;
+  }
+
+  public String readExampleData(final long processDefinitionKey, final String elementId) {
+    final String failureMessagePrefix =
+        String.format(
+            "BPMN Model [processDefinitionKey: '%s', elementId: '%s']",
+            processDefinitionKey, elementId);
```

Comment:
🔧 A user can't extract the process from the process definition key. Instead, we should print the process ID.